lib/posix-process/execve.c

```diff
@@ -0,0 +1,194 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2023, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#include <uk/binfmt.h>
+#include <uk/config.h>
+#include <uk/event.h>
+#include <uk/essentials.h>
+#include <uk/plat/config.h>
+#include <uk/process.h>
+#include <uk/syscall.h>
+#include <uk/thread.h>
+
+#include "process.h"
+
+void execve_arch_execenv_init(struct ukarch_execenv *execenv,
+			      __uptr ip, __uptr sp);
+
+UK_EVENT(POSIX_PROCESS_EXECVE_EVENT);
+
+/* arg0: execenv_new, arg1: stack_old */
+static void __noreturn execve_ctx_switch(long arg0, long arg1)
+{
+	struct ukarch_execenv *execenv_new;
+	struct posix_thread *pthread_parent;
+	struct posix_thread *pthread;
+	struct uk_thread *this_thread;
+	void *stack_old;
+
+	execenv_new = (struct ukarch_execenv *)arg0;
+	stack_old = (void *)arg1;
+
+	this_thread = uk_thread_current();
+	UK_ASSERT(this_thread);
+
+	pthread = tid2pthread(ukthread2tid(this_thread)); /* FIXME */
```

Comment:
Should we assert `pthread` too? :thinking: Or we assume it's fine based on the above assertion?

Reply:
I can add an assertion